baseband/vlbi_base/base.py

```diff
@@ -131,21 +131,21 @@ def __init__(self, fh_raw, header0, nchan, bps, complex_data, thread_ids,
         if frames_per_second is None and sample_rate is None:
             try:
                 frames_per_second = self._get_frame_rate(fh_raw,
-                                                         type(header0))
+                                                         header0)
             except Exception as exc:
-                exc.args += ("the frame rate could not be auto-detected. "
+                exc.args += ("The frame rate could not be auto-detected. "
                              "This can happen if the file has less than "
-                             "one second of data, or because it is "
-                             "corrupted.  Try passing in an explicit "
-                             "'frames_per_second'.",)
+                             "one second of data, is in Mark 4 format and "
```

Comment:
Hmm, I see why you do this, but `vlbi_base` should not refer to subclasses. Maybe just rewrite as "... if the file is too short to determine the frame rate. Try ..."